### PR TITLE
Live "search" API endpoint.

### PR DESCRIPTION
**Pull request by bcipolli**
This implements https://github.com/caciviclab/disclosure-backend/pull/98 on top of #124 . This provides an API endpoint to search cities by name. It is now possible to also search by zip code; I could easily hook that up if desired.

These are raw city names, as parsed from `netfile_agencies.csv`. More work can be done to prettify the names themselves, but that's a separate PR.

image

### CODE REVIEW

File: finance/tests.py [hunk 1]
```diff
@@ -1,2 +1,27 @@
...
+        # Smoke test--make sure there are no errors.
+        call_command('xformnetfilerawdata', verbosity=0,
+                     agencies=test_agency, years=test_year)
+
+        # Check data WESTSAC CLF STO COS
+        self.assertTrue(Ballot.objects.all().count() > 0)
+        self.assertTrue(BallotItem.objects.all().count() > 0)
+        self.assertTrue(BallotItemResponse.objects.all().count() > 0)
+
+        self.assertTrue(IndependentMoney.objects.all().count() > 0)
```
Comment (adborden) [inline finance/tests.py]: Would be nice to have assertion error messages in here so we know which model failed the test.

File: disclosure/urls.py [hunk 2]
```diff
@@ -16,9 +16,7 @@
                                'document_root': settings.STATIC_ROOT,
                                'show_indexes': True,
                            }),
-                       url(r'', include('locality.urls')),
-                       url(r'^locations/(?P<fips_id>[0-9]+)$',
-                           views.location_view),
```
Comment (adborden) [inline disclosure/urls.py:21]: Don't delete this yet, [it still exists](https://github.com/bcipolli/opendisclosure-backend/blob/live-search/disclosure/views.py#L34) as a stub.

Comment (bcipolli) [inline disclosure/urls.py:21]: @adborden Is there a front-end readout of this API endpoint? And is there any list of data stubbed in the front-end code? 

Connecting those up will help understand whether the data this PR lives on top of looks good enough for front-end devs to work with. The "search" one is fine, but any others would really help!

**Comment by bcipolli** [comment]
After further review, I think this is the wrong design. When this API endpoint resides within `locality`, it returns all matching localities--not the ones that have `ballot` or `disclosure` information.

So, looks like this belongs inside the `disclosure` app (like most will).

image

**Comment by bcipolli** [comment]
That's more like it!
image

**Comment by bcipolli** [comment]
Added tests, code coverage up to 88.5% :dancers: Gonna merge and use the test framework to add tests to the other PRs.

**Pull request merged**